### PR TITLE
Photon ID for 2018

### PR DESCRIPTION
**Pull request by trtomei**
From https://twiki.cern.ch/twiki/bin/view/CMS/EgammaRunIIRecommendations
and https://twiki.cern.ch/twiki/bin/view/CMS/MultivariatePhotonIdentificationRun2

Added pt cut > 180 (trigger is at 175), loose and tight ID, and electron veto.

### CODE REVIEW

File: grinder/utils/ids.py [hunk 1]
```diff
@@ -96,15 +96,18 @@ def isLooseTau(pt,eta,decayMode,_id,year):
 pho_id['2017'] = {}
 pho_id['2018'] = {}
 
-#pho_id['2016']['loose_id'] = 'Electron_mvaSpring16GP_WP90'
-#pho_id['2017']['loose_id'] = 'Electron_mvaFall17Iso_WP90'
+pho_id['2018']['loose_id'] = 'Photon_mvaID_WP90'
+pho_id['2018']['tight_id'] = 'Photon_mvaID_WP80'
+pho_id['2018']['eleveto']   = 'Photon_electronVeto'
 
 def isLoosePhoton(pt,eta,year):
```
Comment (mcremone) [inline grinder/utils/ids.py]: add the id and the eleveto as arguments in the function

File: grinder/utils/ids.py [hunk 2]
```diff
@@ -96,15 +96,18 @@ def isLooseTau(pt,eta,decayMode,_id,year):
...
     mask = ~(pt==np.nan)#just a complicated way to initialize a jagged array with the needed shape to True
     if year=='2016':
         mask = (pt>15)*(abs(eta)<2.5)
     elif year=='2017':
         mask = (pt>15)*(abs(eta)<2.5)
+    elif year=='2018':
+        mask = (pt>20)&(abs(eta)<2.5)&(loose_id)&(eleveto)
     return mask
 
 def isTightPhoton(pt,eta,year):
```
Comment (mcremone) [inline grinder/utils/ids.py]: add the id and the eleveto as arguments in the function

File: grinder/utils/ids.py [hunk 3]
```diff
@@ -113,4 +116,6 @@ def isTightPhoton(pt,eta,year):
         mask = (pt>15)*(abs(eta)<2.5)
     elif year=='2017':
         mask = (pt>15)*(abs(eta)<2.5)
+    elif year=='2018':
+        mask = (pt>180)&(abs(eta)<2.5)&(loose_id)&(eleveto) # Trigger threshold is at 175
```
Comment (mcremone) [inline grinder/utils/ids.py]: this should be tight_id instead of loose_id

**Comment by trtomei** [comment]
Done!

**Comment by mcremone** [comment]
Great, thanks

**Pull request merged**